src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```diff
@@ -137,4 +139,13 @@ public boolean isImmutable() {
 	public boolean isParentProperty() {
 		return isParent;
 	}
+
+	@Override
+	public boolean isEntity() {
+
+		TypeInformation<?> typeInformation = getTypeInformation();
```

Comment:
This is exactly what `AbstractPersistentProperty.isEntity()` is doing. See https://github.com/spring-projects/spring-data-commons/blob/48c9297118e18273a0a3dfe3cf2f9a8ffd8fdca7/src/main/java/org/springframework/data/mapping/model/AbstractPersistentProperty.java#L88-L91 and https://github.com/spring-projects/spring-data-commons/blob/48c9297118e18273a0a3dfe3cf2f9a8ffd8fdca7/src/main/java/org/springframework/data/mapping/model/AbstractPersistentProperty.java#L247.

My previous comment should indicate to just use `isEntity()` so you can get rid of the code duplication.

Reply:
aargh, I only looked at the base implementation in `AbstractPersistentProperty` but forgot to check the implementation of the lazy get(); therefore I thought it was different.